Config for pulling Ark overcloud host images

Alex-Welsh opened the pull request:
Updated the Bifrost and pulp config to easily use the overcloud host images stored in Ark.

Currently untested. If CI works with the new image definition, I'll run it through an AUFN

Alex-Welsh committed:
Config for pulling Ark overcloud host images

Updated the Bifrost and pulp config to easily use the overcloud host
images stored in Ark.

markgoddard commented on etc/kayobe/stackhpc-overcloud-host-images.yml:
```diff
@@ -0,0 +1,28 @@
+                                    {{ openstack_release }}/{{ os_distribution }}/{{ os_release }}/\
+                                    {{ 'ofed/' if stackhpc_overcloud_host_image_is_ofed else '' }}\
+                                    {{ stackhpc_overcloud_host_image_version }}/\
+                                    overcloud-{{ os_distribution }}-{{ os_release }}-\
```
trailing slash for non-OFED images?

Alex-Welsh replied:
I don't think there would be a slash, but I have just spotted a trailing dash

Alex-Welsh replied:
I've fixed the stray dash. Do you still think there is an issue with it somewhere?

markgoddard replied:
I meant to say dash

Alex-Welsh committed:
Update release train documentation

Updated the documentation for release train, in particular adding
information on how to use the host images in Ark, and updating the
details surrounding Ubuntu packages.

Alex-Welsh committed:
Fix grep in all-in-one workflow

markgoddard commented on doc/source/configuration/release-train.rst:
```diff
@@ -37,10 +37,12 @@ Configuration
 * Configuration to deploy a local Pulp service as a container on the seed
-* Pulp repository definitions for CentOS Stream 8 and Rocky Linux 8
+* Pulp repository definitions for CentOS Stream 8, Rocky Linux 8/9 and Ubuntu
+  Jammy
```
```suggestion
  Focal
```

Alex-Welsh replied:
Fixed

markgoddard commented on etc/kayobe/kolla/config/bifrost.yml:
```diff
@@ -0,0 +1,8 @@
+---
+# Use prebuilt release train images from Ark.
+{% if stackhpc_download_overcloud_host_images | bool %}
+overcloud_dib_build_host_images: false
```
This is a Kayobe variable and doesn't apply here.

markgoddard replied:
You could override the default in `etc/kayobe/overcloud-dib.yml`, setting it to

```
"{{ os_distribution == 'rocky' and not stackhpc_download_overcloud_host_images | bool }}"
```

Alex-Welsh replied:
Done

markgoddard commented on etc/kayobe/pulp.yml:
```diff
@@ -32,8 +32,14 @@ pulp_proxy_url: "{{ omit }}"
 # StackHPC Pulp server
 
+# The scheme - http/https of the StackHPC Pulp service.
+stackhpc_release_pulp_scheme: "https://"
```
nit: I think strictly the scheme is https

Alex-Welsh replied:
Fixed

markgoddard commented on etc/kayobe/stackhpc-overcloud-host-images.yml:
```diff
@@ -0,0 +1,36 @@
+                                    overcloud-{{ os_distribution }}-{{ os_release }}\
+                                    {{ '-ofed' if stackhpc_overcloud_host_image_is_ofed else '' }}.qcow2"
+
+# Overcloud host image versioning tags
+# These images must be in SMS, since they are used by our AIO CI runners
+stackhpc_centos_8_stream_overcloud_host_image_version: "yoga-20230525T095243"
+stackhpc_rocky_8_overcloud_host_image_version: "yoga-20230629T135322"
+stackhpc_rocky_9_overcloud_host_image_version: "yoga-20230515T145140"
+stackhpc_ubuntu_focal_overcloud_host_image_version: "yoga-20230609T120720"
+stackhpc_ubuntu_jammy_overcloud_host_image_version: "yoga-20230609T120720"
```
It will be easier to automate updates to these versions if they are in a separate file, like pulp-repo-versions.yml

Alex-Welsh replied:
I've moved them across to pulp-host-image-versions.yml

markgoddard commented on releasenotes/notes/pull-stackhpc-host-images-a623e4ab2d2a9e5b.yaml:
```diff
@@ -0,0 +1,6 @@
+---
+features:
+  - |
+    Prebuild overcloud host images can now be pulled from Ark using the
```
```suggestion
    Prebuilt overcloud host images can now be pulled from Ark using the
```

Alex-Welsh replied:
Done

markgoddard commented on doc/source/configuration/host-images.rst:
```diff
@@ -4,6 +4,44 @@
+* Ubuntu Jammy 22.04
+
+The image to download is selected automatically using the ``os_distribution``
+and ``os_release`` variables.
```
Might be nice to mention that the images are versioned and where the versions are stored.

Alex-Welsh replied:
Done

Alex-Welsh committed:
Update host image pulling changes after review

markgoddard reviewed (approved):
Nice!
